src/main/webapp/app/course/competencies/competency-card/competency-card.component.html

```diff
@@ -22,45 +22,47 @@
-                    @if (competency.optional) {
+                    @if (competency()?.optional) {
                         <span id="optional-badge" class="badge text-white bg-warning" jhiTranslate="artemisApp.competency.optional"></span>
                     }
                 </h4>
-                @if (competency.description) {
-                    <p class="m-0 markdown-preview" [innerHTML]="competency.description | htmlForMarkdown"></p>
+                @if (competency()?.description) {
+                    <p class="m-0 markdown-preview" [innerHTML]="competency()?.description | htmlForMarkdown"></p>
                 }
-                @if (isPrerequisite && competency.linkedCourseCompetency?.course) {
+                @if (isPrerequisite() && competency()?.linkedCourseCompetency?.course) {
                     <div>
-                        @if (competency.linkedCourseCompetency!.course!.title) {
-                            <span class="badge bg-primary">{{ competency.linkedCourseCompetency!.course!.title! }}</span>
+                        @if (competency()!.linkedCourseCompetency!.course!.title) {
+                            <span class="badge bg-primary">{{ competency()!.linkedCourseCompetency!.course!.title! }}</span>
                         }
-                        @if (competency.linkedCourseCompetency!.course!.semester) {
-                            <span class="badge bg-secondary">{{ competency.linkedCourseCompetency!.course!.semester! }}</span>
+                        @if (competency()!.linkedCourseCompetency!.course!.semester) {
+                            <span class="badge bg-secondary">{{ competency()!.linkedCourseCompetency!.course!.semester! }}</span>
                         }
                     </div>
                 }
                 <ng-content />
             </div>
-            @if (competency.softDueDate) {
+            @if (competency()?.softDueDate) {
                 <div class="col-sm col py-2" style="max-width: fit-content">
                     <span style="vertical-align: middle" jhiTranslate="artemisApp.competency.competencyCard.softDueDate"></span>
                     <span class="badge" [ngClass]="softDueDatePassed && !isMastered ? 'bg-danger' : 'bg-success'" id="date-badge">
-                        {{ competency.softDueDate! | artemisTimeAgo }}
+                        {{ competency()!.softDueDate! | artemisTimeAgo }}
                     </span>
                 </div>
             }
-            <div class="col-sm col text-end py-1 px-2" style="max-width: 95px">
-                <jhi-competency-rings [progress]="progress" [mastery]="mastery" [hideProgress]="hideProgress" />
-            </div>
+            @if (!noProgressRings()) {
+                <div class="col-sm col text-end py-1 px-2" style="max-width: 95px">
+                    <jhi-competency-rings [progress]="progress" [mastery]="mastery" [hideProgress]="hideProgress()" />
+                </div>
+            }
```

Comment:
This is nit-picky, but can we avoid the double negation by having a `showProgressRings` input that is true by default?

Reply:
I thought about this as well, but at least I find it more intuitive to set `noProgressRings` to `true` to mark an exception in the normal behavior...